File: Assets/PythonMCP/modules/instantiate_prefab_module.py
import requests
import json
from typing import Dict

class InstantiatePrefabModule:
    def __init__(self, base_url: str):
        self.base_url = base_url
    
    def execute(self, prefab_asset_path: str, parent_path: str = "", name_override: str = "") -> Dict:
        """Инстанцирует префаб на сцене"""
        try:
            if not prefab_asset_path:
                return {
                    "success": False,
                    "action": "instantiate_prefab",
                    "error": "prefab_asset_path is required"
                }
            
            data = {
                "prefabPath": prefab_asset_path,
                "parentPath": parent_path,
                "name": name_override
            }
            
            response = requests.post(
                f"{self.base_url}/objects/instantiate/prefab",
                json=data
            )
            response.raise_for_status()
            result = response.json()
            
            return {
                "success": result.get("success", False),
                "action": "instantiate_prefab",
                "data": result,
                "error": result.get("error")
            }
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "instantiate_prefab",
                "error": f"Request error: {str(e)}"
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "instantiate_prefab",
                "error": f"JSON decode error: {str(e)}"
            }

File: Assets/Editor/SceneAPI/Modules/Prefabs/InstantiatePrefabModule.cs
using System;
using System.Net;
using Newtonsoft.Json;
using UnityEditor;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class InstantiatePrefabModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);

                string prefabPath = data?.prefabPath;
                string parentPath = data?.parentPath ?? "";
                string nameOverride = data?.name ?? "";

                if (string.IsNullOrEmpty(prefabPath))
                {
                    return JsonConvert.SerializeObject(new { success = false, error = "prefabPath is required" });
                }

                GameObject prefab = AssetDatabase.LoadAssetAtPath<GameObject>(prefabPath);
                if (prefab == null)
                {
                    return JsonConvert.SerializeObject(new { success = false, error = $"Prefab not found at path: {prefabPath}" });
                }

                GameObject instance = PrefabUtility.InstantiatePrefab(prefab) as GameObject;
                if (instance == null)
                {
                    return JsonConvert.SerializeObject(new { success = false, error = "Failed to instantiate prefab" });
                }

                if (!string.IsNullOrEmpty(nameOverride))
                {
                    instance.name = nameOverride;
                }

                if (!string.IsNullOrEmpty(parentPath))
                {
                    var parent = GameObjectUtilities.FindGameObjectByPath(parentPath);
                    if (parent != null)
                    {
                        instance.transform.SetParent(parent.transform);
                    }
                }

                string fullPath = string.IsNullOrEmpty(parentPath) ? instance.name : $"{parentPath}/{instance.name}";

                return JsonConvert.SerializeObject(new
                {
                    success = true,
                    path = fullPath,
                    instanceId = instance.GetInstanceID(),
                    message = $"Prefab instantiated: {instance.name}"
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new { success = false, error = $"Error instantiating prefab: {ex.Message}" });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new System.IO.StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}